app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -1,94 +1,77 @@
 module ManageIQ::Providers
   module Hawkular
-    class MiddlewareManager::RefreshParser
+    class Inventory::Parser::MiddlewareManager < ManagerRefresh::Inventory::Parser
       include ::Hawkular::ClientUtils
 
-      def self.ems_inv_to_hashes(ems, options = nil)
-        new(ems, options).ems_inv_to_hashes
-      end
-
-      def initialize(ems, _options = nil)
-        @ems = ems
-        @eaps = []
-        @data = {}
+      def initialize
         @data_index = {}
       end
 
-      def ems_inv_to_hashes
-        # the order of the method calls is important here, because they make use of @eaps and @data_index
+      def parse
+        # the order of the method calls is important here, because they make use of @data_index
```

Comment:
actually the order should not matter anymore and you should not need `@data_index` anymore.
Are you about to remove the dependency on `@data_index` ?